### ISSUE TITLE
Maya: Switch Asset doesn't update namespace

### ISSUE THREAD

**Issue by BigRoy** [pypeclub/OpenPype#2526]
**Describe the bug**

When loading e.g. `pointcacheFoo` in asset `nugget` the loaded namespaces by default is `nugget_pointcacheFoo_01_`.
However when I switch it to `pointcacheBar` in asset `hero` it would keep the original namespace `nugget_pointcacheFoo_01_`. 

This makes it quite confusing when at the outliner as I'm not sure `nugget_pointcacheFoo_01_` is actually what it's telling me to be.

**To Reproduce**

1. Go to OpenPype > Load...
2. Load something
3. Go to OpenPype > Manage..
4. Richt click the loaded content, click Switch Asset
5. Switch it to something else.

Note how the namespace hasn't changed.

**Expected behavior**

I'd like to propose the namespace/naming to update along with the switch.

[Here's an implementation example as we originally had it in our Colorbleed config](https://github.com/Colorbleed/colorbleed-config/commit/3970232e95a20d48ce6b6b1ca656f7f399914659).

**Comment by tokejepsen** [comment]
We often switch caches when reusing setups between shots, in which case we dont want to destroy any connections. Guessing this would be non-destructive?

**Comment by BigRoy** [comment]
> We often switch caches when reusing setups between shots, in which case we dont want to destroy any connections. Guessing this would be non-destructive?

In our tests reference edits would be maintained.

However it **would** cause an issue if you:
1. Set up **animation** shot 010
2. Create **animation** shot 020 by taking animation 010 and switching assets. (not sure if you'd do that on layout/animation already?)
3. Publish **animation** shot 010 & 020
4. Set up **lighting** shot 010, loading publishes of animation of shot 010
5. Create **lighting** shot 020 by taking lighting 010 and switching assets.

The reference edits from 010 to 020 would preserve in the way that upon changing the reference namespace Maya knows how to update the reference edits there. However, if 020 would have loaded content where INSIDE the referenced content namespaces had altered then yes, Maya wouldn't find them upon switching in that particular case. So it would fail on namespaces in the loaded content different. So it **could fail with nested namespaces.**

TL;DR changing namespace on the reference itself is no problemo, but once you might publish that again where you rely on a dag path including that namespace. Then yes, later that namespace would be different across shots.
